lisa/sut_orchestrator/baremetal/bootconfig.py

```diff
@@ -0,0 +1,178 @@
+class PxeBoot(BootConfig):
+    def __init__(
+        self,
+        runbook: PxeBootSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.pxe_runbook: PxeBootSchema = self.runbook
+        self._log = get_logger("pxe_boot", self.__class__.__name__)
+        self._boot_label = "label lisav3"
+        self._boot_dir = "/var/lib/tftpboot/"
+        self._boot_config = self._boot_dir + "pxelinux.cfg/"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "pxe_boot"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return PxeBootSchema
+
+    def set_boot_config(self) -> None:
+        try:
+            self._connect_to_pxe_server()
+        except Exception as ex:
+            self._log.debug(
+                f"failed to connect {self.pxe_runbook.connection}",
+                exc_info=ex,
+            )
+
+        node_name = f"{self.pxe_runbook.node_setup}{self.pxe_runbook.node_number}"
+        host_ref = f"host {node_name}"
+        pxe_boot_config_path = self._get_dhcp_info(node_name, host_ref)
+        if not pxe_boot_config_path:
+            raise LisaException(f"Failed to find DHCP entry for {node_name}")
+
+        boot_image = self.pxe_runbook.image_source.replace(
+            self._boot_dir,
+            "",
+        ).strip("/")
+        boot_entry = f"kernel {boot_image}"
+
+        # Delete it if one is there, so to always insert at the top
+        self.pxe_server.execute(
+            cmd=f"sed -i '/^{self._boot_label}$/,/^$/d' " f"{pxe_boot_config_path}",
+            expected_exit_code_failure_message=(
+                f"Failed to delete previous boot entry on {node_name}"
+            ),
+            expected_exit_code=0,
+            sudo=False,
+        )
+
+        self._log.debug(
+            "Adding boot entry for LISA at "
+            f"{pxe_boot_config_path}, "
+            f"pointing to {boot_entry}"
+        )
+
+        entry = self.pxe_runbook.kernel_append
+        append = f"\\n    append {entry}" if entry else ""
+        label = self._boot_label
+        self.pxe_server.execute(
+            cmd=f"sed -i '/^menu.*/a \\\\n{label}\\n    {boot_entry}{append}' "
+            f"{pxe_boot_config_path}",
+            expected_exit_code_failure_message=(
+                f"Failed to create new boot entry on {node_name}"
+            ),
+            expected_exit_code=0,
+            sudo=False,
+        )
+
+        self._log.debug(
+            "Added boot entry for LISA at "
+            f"{pxe_boot_config_path}, "
+            f"pointing to {boot_entry}"
+        )
+
+    def _connect_to_pxe_server(self) -> None:
+        assert self.pxe_runbook.connection, "connection is required for pxe_server"
+        self.pxe_runbook.connection.name = "pxe_server"
+        self.pxe_server = quick_connect(
+            self.pxe_runbook.connection, logger_name="pxe_server"
+        )
+
+    def _get_dhcp_info(self, node_name: str, host_ref: str) -> str:
```

Comment:
This method output the pxe config path. So the file name should be like `_get_pxe_config_path`.